app/controllers/projects_controller.rb

```diff
@@ -0,0 +1 @@
+before_filter :authenticate_user!, :except => [:show, :index]
```

Comment:
Do we need to authenticate before show and index ?

Reply:
we do not need to authenticate. However, In this code, It is permitting except for show and index.